release-notes.txt: add 2025.01 release notes (#21183)
RIOT-OS/RIOT: release-notes.txt

found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
light-weight 32-bit processors.

RIOT is based on the following design principles: energy-efficiency, soft
real-time capabilities, small memory footprint, modularity, and uniform API
access, independent of the underlying hardware (with partial POSIX compliance).

RIOT is developed by an international open-source community which is
independent of specific vendors (e.g. similarly to the Linux community) and is
licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
business models around the free open-source software platform provided by RIOT.

About this release
==================

The 2025.01 release includes:

## List of notable changes here ##

144 pull requests, composed of 280 commits, have been merged since the
last release, and 0 issues have been solved. 32 people contributed with
code in 65 days. 665 files have been touched with 75661 (+) insertions and
6329 deletions (-).

Notations used below
====================

    + means new feature/item
    * means modified feature/item
    - means removed feature/item

New features and changes

> ========================

Reviewer: `GOVERNANCE.md` is not mentioned, as far as I can see.

Reviewer 2: That would also be the spotlight feature to point out first in the release notes IMO.

Author: Done

Core
----

+ core/cib: add const to cib_peek() (#21130)
+ core/mutex: add `mutex_init_locked()` (#20954)

> * core/msg: re-enable IRQs before printing for highlevel_stdio (#21047)

Reviewer 3: Yes, I'd classify this as a bugfix.

> * core/msg: re-enable IRQs before printing for highlevel_stdio (#21047)

Author: done

System Libraries
----------------

+ sys/posix/pthread: Add pthread_attr_getstack and
  pthread_attr_setstack (#20934)
+ sys/stdio_nimble: add version note to README (#21085)
+ sys/stdio_uart: add stdio_clear_stdin (#19837)
* suit: move CoAP endpoints to the example (#21045)
* sys/chunked_ringbuffer: discard stale chunk when starting a new one (#21073)
* sys/psa_crypto: split algorithm definitions (#20906)
* sys/psa_crypto: Update key headers (#21046)
* sys/shell: cmds_json builtin command (#20964)
* sys/shell: reduce overhead of XFA shell commands (#20958)
* sys/stdio: stdio_available is always false with missing stdin module (#21040)
* sys/usb: Use luid_base for stable USB serials (#20944)
* sys/usb_cdc_acm_stdio: only submit and flush for non-empty buffer (#20986)

> * treewide: Update Rust dependencies (#21133)

Reviewer 4: ```suggestion
* treewide: Update Rust dependencies, adding SPI and UART support (#21133)
```

* Arduino support layer: several fixes (#21039)
* boards/adafruit-metro-m4-express: Minor improvements (#21041)
* boards/nrf52*dk: Enable Pinreset after Flashing (#20965)
* cpu/nRF52: Always recover Device before Flashing (#20970)
* makefiles/serial: better MOST_RECENT_PORT=1 for CDC ACM (#20951)

CPU
---

+ dist/tools/esptools: add support for installing esp8266 toolchain (#21094)
+ sys/string_utils: add string_writer helper (#20843)
* cpu/cc2538: use RX FIFO count to determine CRC OK location (#20956)
* cpu/cortexm_common: Make cpu.h IWYU clean (#21050)
* cpu/esp32/stdio_usb_serial_jtag: Fix to ESP32 stdio usb serial
  hanging if it receives data too quickly (#20972)
* cpu/sam0_common: flashpage: disable cache while writing (#21043)
* netdev_tap: port to `netdev_new_api` (#21015)
* sys/psa_crypto: one-shot Chacha20 support (#20720)

Device Drivers
--------------

* drivers/atwinc15x0: port to `netdev_new_api` (#21020)
* drivers/dose: fix poweroff (#21074)
* drivers/dose: port to netdev_new_api (#20991)
* drivers/ethos: port to netdev_new_api (#21016)
* drivers/periph_init: initialize watchdog first (#21025)
* drivers/slipdev: implement NETDEV_EVENT_TX_COMPLETE event (#20975)

> * drivers/slipdev: port to netdev_new_api (#20978)

Author: mentioned in the spotlight

RIOT-2025.01 - Release Notes
============================
RIOT is a multi-threading operating system which enables soft real-time
capabilities and comes with support for a range of devices that are typically
found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
light-weight 32-bit processors.

RIOT is based on the following design principles: energy-efficiency, soft
real-time capabilities, small memory footprint, modularity, and uniform API
access, independent of the underlying hardware (with partial POSIX compliance).

RIOT is developed by an international open-source community which is
independent of specific vendors (e.g. similarly to the Linux community) and is
licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
business models around the free open-source software platform provided by RIOT.

About this release
==================

The 2025.01 release includes:

> More work done to the legacy `netdev` API to the update version `new_netdev_api`
> (#20978, #21012, #21015, #21016, #21020, #21028, #21091, #20991). This means
> that if you are still using the legacy `netdev` API, start thinging about
> updating!

Reviewer 3: ```suggestion
More work done to update the legacy `netdev` API to its new version `new_netdev_api`
(#20978, #21012, #21015, #21016, #21020, #21028, #21091, #20991). This means
that if you are still using the legacy `netdev` API, start thinking about
updating!
```

* sys/shell: reduce overhead of XFA shell commands (#20958)
* sys/stdio: stdio_available is always false with missing stdin module (#21040)
* sys/usb: Use luid_base for stable USB serials (#20944)
* sys/usb_cdc_acm_stdio: only submit and flush for non-empty buffer (#20986)
* treewide: Update Rust dependencies, adding SPI and UART support (#21133)
* usbus/cdc_ecm: port to netdev_new_api (#21028)

Networking
----------

+ gnrc/ipv6_auto_subnets: add option to derive subnet prefix from EUI (#21080)
+ sys/event: add assertion that event has a handler (#20847)
+ sys/net/gnrc_pktbuf_static: add double free detection (#20974)
* gnrc_netif: netdev_new_api implies TX end irq, no need to check it (#20993)
* nanocoap/sock: re-try if wrong block was received (#20952)
* sys/net/gcoap: get rid of API abuse (#21125)
* sys/net/gcoap: reduce insanity of hack (#20945)
* sys/net/gnrc_pktbuf: detect use after free if canary is in metadata (#21000)
* sys/net/nanocoap: flush stale responses before sending request (#21122)
* sys: net: dns: use Quad9 resolver (#21120)

Packages
--------

+ Feature/wakaama update on new version (#20930)
* pkg/relic: bump version to 0.7.0 (#21134)

Boards
------

> + boards/same51-curiosity-nano: add support for the board (#21111)

Reviewer 3: we could maybe mention this new board as one of the highlights

Author: done

+ cpu/saml21: add compatibility headers for vendor files migration (#20850)
+ feather-nrf52840: add config for NeoPixel LED (#21017)
+ saul: initial import of saul_bat_voltage module (#21018)
* Arduino support layer: several fixes (#21039)
* boards/adafruit-metro-m4-express: Minor improvements (#21041)
* boards/nrf52*dk: Enable Pinreset after Flashing (#20965)
* cpu/nRF52: Always recover Device before Flashing (#20970)
* makefiles/serial: better MOST_RECENT_PORT=1 for CDC ACM (#20951)

CPU
---

+ dist/tools/esptools: add support for installing esp8266 toolchain (#21094)
+ sys/string_utils: add string_writer helper (#20843)
* cpu/cc2538: use RX FIFO count to determine CRC OK location (#20956)
* cpu/cortexm_common: Make cpu.h IWYU clean (#21050)
* cpu/esp32/stdio_usb_serial_jtag: Fix to ESP32 stdio usb serial
  hanging if it receives data too quickly (#20972)
* cpu/sam0_common: flashpage: disable cache while writing (#21043)
* netdev_tap: port to `netdev_new_api` (#21015)

> * sys/psa_crypto: one-shot Chacha20 support (#20720)

Reviewer 3: This should rather be in System Libraries together with the other `sys/psa_crypto` things.

Author: done

RIOT is a multi-threading operating system which enables soft real-time
capabilities and comes with support for a range of devices that are typically
found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
light-weight 32-bit processors.

RIOT is based on the following design principles: energy-efficiency, soft
real-time capabilities, small memory footprint, modularity, and uniform API
access, independent of the underlying hardware (with partial POSIX compliance).

RIOT is developed by an international open-source community which is
independent of specific vendors (e.g. similarly to the Linux community) and is
licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
business models around the free open-source software platform provided by RIOT.

About this release
==================

The 2025.01 release includes:

More work done to update the legacy `netdev` API to its new version `new_netdev_api`
(#20978, #21012, #21015, #21016, #21020, #21028, #21091, #20991). This means
that if you are still using the legacy `netdev` API, start thinking about
updating!

We documented the community processes in a `GOVERNANCE.md` file.

> This new file now clearly states all previously undocumented community procedures and governance, and defines community roles, to further our goals of openness and transparency within the community.

Reviewer 5: ```suggestion
This new file now clearly states all previously undocumented community procedures and
governance, and defines community roles, to further our goals of openness and transparency
within the community.
```

  hanging if it receives data too quickly (#20972)
* cpu/sam0_common: flashpage: disable cache while writing (#21043)
* netdev_tap: port to `netdev_new_api` (#21015)

Device Drivers
--------------

* drivers/atwinc15x0: port to `netdev_new_api` (#21020)
* drivers/dose: fix poweroff (#21074)
* drivers/dose: port to netdev_new_api (#20991)
* drivers/ethos: port to netdev_new_api (#21016)
* drivers/periph_init: initialize watchdog first (#21025)
* drivers/slipdev: implement NETDEV_EVENT_TX_COMPLETE event (#20975)
* drivers/slipdev: port to netdev_new_api (#20978)
* drivers/ws281x: esp32 neopixel driver to set the rmt symbol high/low
  lengths at init instead of every write (#21068)

Documentation
-------------

+ boards/nucleo-l073rz: add MCU table to doc page (#21102)
+ boards/nucleo-l433rc: add pinout to documentation page (#20947)
+ examples: add TCP echo server & client from documentation (#16739)
+ pkg/u8g2: add to the displays doc group (#20953)
* doc: Improvements around Code of Conduct (#21071)
* doc: move "managing a release" from wiki (#21065)
* MAINTAINING.md/doc: move links from wiki to doc (#21066)
* sys/net/app/cord: update doc (#20921)

> * GOVERNANCE.md: remove “Community Processes” in favor of “GOVERNANCE.md”
> (#21067)

Reviewer 5: ```suggestion
* GOVERNANCE.md: remove “Community Processes” in favor of “GOVERNANCE.md”
  (#21067)
```

RIOT-2025.01 - Release Notes
============================
RIOT is a multi-threading operating system which enables soft real-time
capabilities and comes with support for a range of devices that are typically
found in the Internet of Things: 8-bit and 16-bit microcontrollers as well as
light-weight 32-bit processors.

RIOT is based on the following design principles: energy-efficiency, soft
real-time capabilities, small memory footprint, modularity, and uniform API
access, independent of the underlying hardware (with partial POSIX compliance).

RIOT is developed by an international open-source community which is
independent of specific vendors (e.g. similarly to the Linux community) and is
licensed with a non-viral copyleft license (LGPLv2.1), which allows indirect
business models around the free open-source software platform provided by RIOT.

About this release
==================

The 2025.01 release includes:

More work done to update the legacy `netdev` API to its new version `new_netdev_api`
(#20978, #21012, #21015, #21016, #21020, #21028, #21091, #20991). This means
that if you are still using the legacy `netdev` API, start thinking about
updating!

> Documented the community processes in a `GOVERNANCE.md` file, clearly stating
> all previously undocumented procedures to further our goals of openness and
> transparency within the community.

Reviewer: ```suggestion
We documented the community processes in a `GOVERNANCE.md` file. 
This new file now clearly states all previously undocumented community procedures and governance, and defines community roles, to further our goals of openness and transparency within the community.
```
